base/applications/kbswitch/kbswitch.c

```diff
@@ -55,12 +55,23 @@ vDbgPrintExWithPrefix(IN PCCH Prefix,
                       IN PCCH Format,
                       IN va_list ap)
 {
-    CHAR Buffer[512];
-    SIZE_T PrefixLength = strlen(Prefix);
-    strncpy(Buffer, Prefix, PrefixLength);
-    _vsnprintf(Buffer + PrefixLength, _countof(Buffer) - PrefixLength, Format, ap);
-    Buffer[_countof(Buffer) - 1] = ANSI_NULL; /* Avoid buffer overrun */
-    OutputDebugStringA(Buffer);
+    static CHAR s_szBuffer[256] = "";
```

Comment:
You're now making the function not thread-safe (in case there are more than one thread in kbswitch that would use it).

Reply:
`OutputDebugString` generates a line break for each output.

Reply:
I'm debugging using `gdb`. `gdb` generated line break like this:
![image](https://github.com/user-attachments/assets/a96a604c-5550-41bf-bef7-f2b9f6f71d59)

Reply:
"warning:" lines are broken.

Reply:
I know, but didn't you want to remove dependency on ntdll used just because of the DbgPrint usages?

Reply:
As in 'Revert 57981ed and later changes'.